keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,687 @@
+
+This KEP proposes a mechanism for cluster administrators to mark a request in a
+ResourceClaim or ResourceClaimTemplate with an admin access flag. This flag
+allows privileged access to devices, enabling administrative tasks without
+compromising security. Access to this mode is restricted to users authorized to
+create ResourceClaim or ResourceClaimTemplate objects in namespaces marked with
+the DRA admin label, ensuring that non-administrative users cannot misuse this
+feature.
+
+Note: Administrator workloads should not interfere with the application
+workloads. If that cannot be supported due to hardware restrictions or that the
+administrator workload is too demanding, then administrators should request
+exclusive access for specific admin task such that no other claims are
+allocated.
+
+## Motivation
+
+The current Dynamic Resource Allocation (DRA) feature enables users to request
+resources but lacks a mechanism for cluster administrators to gain privileged
+access to devices already in use by other users. This limitation restricts
+administrators from performing tasks like:
+
+- Monitoring device health or status.
+
+- Performing diagnostics and troubleshooting on devices shared across users.
+
+Enabling such administrative tasks without compromising security is critical to
+cluster operations. Non-admin users gaining access to devices in use by others
+could result in:
+
+- Unauthorized access to confidential information about device usage.
+
+- Potential conflicts or misuse of shared hardware.
+
+As the adoption of DRA expands, the inability of administrators to perform
+privileged device introspection becomes a bottleneck for cluster operations,
+particularly in shared environments where devices are critical resources.
+
+### Goals
+
+- Provide a privileged mode for cluster administrators to access devices already
+  in use.
+
+- Ensure that only authorized users can use the privileged mode.
+
+- Maintain security and prevent unauthorized access to devices.
+
+### Non-Goals
+
+- Redesign DRA.
+
+- Extend privileged mode access to non-admin users.
+
+- Define how administrators use the data accessed through privileged mode.
+
+- Define which specific additional permissions are enabled by DRA drivers when
+  this privileged mode is used.
+
+## Proposal
+
+### DRAAdminAccess Overview
+
+This proposal enhances the `DRAAdminAccess` feature to enable cluster
+administrators to mark requests in `ResourceClaim` and `ResourceClaimTemplate`
+objects as privileged. This feature includes:
+
+1. Admin Access Flag:
+
+   A field in `ResourceClaim` and `ResourceClaimTemplate` specifications to
+   indicate the request is in admin mode.
+
+   Example:
+
+   ```yaml
+   apiVersion: resource.k8s.io/v1beta1
+   kind: ResourceClaim
+   metadata:
+     name: admin-resource-claim
+     namespace: admin-namespace
+   spec:
+     resourceClassName: admin-resource-class
+     adminAccess: true
+   ```
+
+1. Namespace Label for DRA Admin Mode:
```

Comment:
it would be helpful to list out the assumptions / limitations of the "label namespace to give access" approach

1. assumption: it is not important to subdivide admin access to different types of devices
2. assumption: ability to label (or relabel) namespaces is not granted to low-privileged namespace-constrained users